cluster_orchestrator/cluster-manager/mongodb_client.py

```diff
@@ -253,26 +268,30 @@ def mongo_update_jobs_status(TIME_INTERVAL):
         try:
             updated = False
             for instance in range(len(job["instance_list"])):
-
                 last_time_job_was_modified = job["instance_list"][instance].get(
                     "last_modified_timestamp", datetime.now().timestamp()
                 )
                 job_is_inactive = last_time_job_was_modified < (
-                    datetime.now().timestamp() - TIME_INTERVAL
+                    datetime.now().timestamp() - time_interval
                 )
-                job_status = job["instance_list"][instance].get("status", 0)
-                job_is_not_completed = job_status != "COMPLETED"
-                job_is_not_scheduled = job_status not in ["NODE_SCHEDULED", "CLUSTER_SCHEDULED"]
-
-                if job_is_inactive and job_is_not_scheduled and job_is_not_completed:
+                job_status = (
+                    convert_to_status(job["instance_list"][instance].get("status"))
+                    or LegacyStatus.LEGACY_0
+                )
+                if (
+                    job_is_inactive
+                    and job_status not in PositiveSchedulingStatus
+                    and job_status != DeploymentStatus.COMPLETE
+                ):
                     print("Job is inactive: " + str(job.get("job_name")))
-                    job["instance_list"][instance]["status"] = "FAILED"
-                    status = "FAILED"
+                    # TODO What is the difference between this status and the prev job_status ?
```

Comment:
Just a minor clarification: what do you mean with _previous job_status_? I believe it's coherent if it update the job status from **Cluster Scheduled/Worker Scheduled** to **Failed** (_despite this should not happen considering the 'COMPLETED' state machine diagram_)

Reply:
The previous job_status is simply the "other status" from a few lines above the "status" 
![image](https://github.com/oakestra/oakestra/assets/65814168/7c33ebfd-dec6-4fda-8780-6134440022c7)

I had another look and managed to resolve the TODO. Now I simply create a new_job_status = FAILED and work with it.

Reply:
You're right, more clear now